app/scripts/lib/PatchStore.ts

```diff
@@ -47,15 +48,31 @@ export class PatchStore {
   }
 
   private _onStateChange({
-    oldState,
     newState,
+    oldState,
+    patches: eventPatches,
   }: {
     controllerKey: string;
-    oldState: Record<string, unknown>;
-    newState: Record<string, unknown>;
+    oldState: Record<string, Json>;
+    newState: Record<string, Json>;
+    patches?: Patch[];
   }) {
-    const sanitizedNewState = sanitizeUIState(newState);
-    const patches = this._generatePatches(oldState, sanitizedNewState);
+    const sanitizedNewState = eventPatches
```

Comment:
Nit: The order of operations here seems potentially confusing. We're assigning `newState` to the variable `sanitizedNewState` here in the case where `eventPatches` exists, which could be dangerously misleading (it's not sanitized!).

This is OK in this case because the variable ends up not used, and we _do_ want to skip the sanitize operation if it's not needed.

But I could easily see a future refactor resulting in _unsanitized_ state being passed to the UI via this `sanitizedNewState` parameter.

We could avoid this by restructuring this into two distinct branches, and having no `sanitizedNewState` variable for the branch with patches.

Reply:
Done.